### ISSUE TITLE
Support customizing containerd namespace

### ISSUE THREAD

**Issue by ttc0419** [moby/moby#47957]
### Description

By default, docker engine uses moby containerd namespace, however other runtime like kubelet uses k8s.io and there's no way to change it. Please consider support customizing containerd namespace so users can have a unified workspace.

**Comment by vvoland** [comment]
There already is a daemon command line flag that allows to specify the containerd namespace to use:

```bash
$ dockerd --help

Usage:  dockerd [OPTIONS]

A self-sufficient runtime for containers.

Options:
...
      --containerd-namespace string           Containerd namespace to use (default "moby")
```

**Comment by ttc0419** [comment]
@vvoland Thanks for the info, I can list images now but I still cannot list containers running using the ```docker ps``` command, what might be the reason?

**Comment by vvoland** [comment]
Right, it only affects the images as docker doesn't use the containerd as a source of truth for the containers, as it has its own containers db, so you won't be able to see any containers started outside of docker in `docker ps`.

I think we want to eventually use the containerd as a source of truth for containers too, but that's not a priority right now.

**Comment by huapox** [comment]
+1
